core/src/main/java/com/netflix/conductor/core/execution/tasks/Join.java

```diff
@@ -66,11 +67,26 @@ public boolean execute(
             if (hasFailures) {
                 break;
             }
+
+            // check for optional task failures
+            if (forkedTask.getWorkflowTask().isOptional()
+                    && taskStatus == TaskModel.Status.COMPLETED_WITH_ERRORS) {
+                optionalTaskFailures
+                        .append(
+                                String.format(
+                                        "%s/%s",
+                                        forkedTask.getTaskDefName(), forkedTask.getTaskId()))
+                        .append(" ");
+            }
         }
-        if (allDone || hasFailures) {
+        if (allDone || hasFailures || optionalTaskFailures.length() > 0) {
             if (hasFailures) {
                 task.setReasonForIncompletion(failureReason.toString());
                 task.setStatus(TaskModel.Status.FAILED);
+            } else if (optionalTaskFailures.length() > 0) {
+                task.setStatus(TaskModel.Status.COMPLETED_WITH_ERRORS);
+                optionalTaskFailures.append("completed with errors");
```

Comment:
nit:
```java
optionalTaskFailures.append(" completed with errors");
```

Reply:
There is a space already appended here: https://github.com/Netflix/conductor/pull/2974/files#diff-ebc515b9038973f4691b79e4dbbfec232800c5a87e95cc2ddbd12033b0fd2926R79 so I am omit the space here :)